### ISSUE TITLE
Diagnostic logger change order of args

### ISSUE THREAD

**Issue by bruno-garcia** [getsentry/sentry-dotnet#699]
The `LogError` should take `Exception` as the first arg. So it doesn't get mixed up with the `<TArg` overloads.
There shouldn't be a nullable Exception parameter, either takes an exception or it doesnt.

**Comment by SimonCropp** [comment]
@bruno-garcia it looks like this is done?

```
       public static void LogError(this ILogger logger, Exception exception, string message, params object[] args)
        {
            logger.Log(LogLevel.Error, exception, message, args);
        }

```

**Comment by bitsandfoxes** [comment]
Closed by [#2715](https://github.com/getsentry/sentry-dotnet/pull/2715)
